### PR TITLE
Feature: optional public RPC and block explorer Chain params

### LINKED ISSUE #1020: [Bug]: addNewChain always sets the app's provider RPC URL
### Current Behavior

In `onboard.setChain`, when the user doesn't have that chain in their wallet, onboard suggests to add it.

To achieve that, [it passes the app's provider configuration](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard/packages/core/src/chain.ts#L71) to the wallet.

In a typical web3 app, that uses a custom Infura endpoint with an individual project id, it's not desirable that the user has this custom RPC URL in their wallet.

### Expected Behavior

There should be a way to specify an RPC URL to give to the wallet when adding a new chain. That would typically be a public RPC.

### Steps To Reproduce

* Use MetaMask
* Remove Polygon if you have it in your MetaMask already
* Initialize onboard with a custom Polygon RPC
* Call `onboard.setChain({ chainId: 137 })`
* Observe that it offers the user to install Polygon with that RPC

### Onboard Version

2

### Node Version

16

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Anything else?

I would suggest to add a new field in the Chain configuration, `publicRpc`, and use that when calling `wallet_addEthereumChain`. Or maybe just a new param for the `setChain` method.

### PR DESCRIPTION
**Pull request by katspaugh**
### Description
Resolves #1020 and #948.

As outlined in #1020, it's sometimes necessary to pass a custom (typically public) RPC when adding a new chain to the user's wallet.

### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [ ] The box that allows repo maintainers to update this PR is checked
- [ ] I tested locally to make sure this feature/fix works
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/core",
-  "version": "2.2.9-alpha.4",
+  "version": "2.2.9-alpha.5",
```
Comment (katspaugh) [inline packages/core/package.json:3]: I'm not sure which version would be correct here.
`2.2.9-alpha.5` or `2.2.10-alpha.4`?

File: packages/demo/src/App.svelte [hunk 2]
```diff
@@ -269,7 +269,7 @@
     <button on:click={() => onboard.setChain({ chainId: '0x4' })}
       >Set Chain to Rinkeby</button
     >
-    <button on:click={() => onboard.setChain({ chainId: '0x89' })}
+    <button on:click={() => onboard.setChain({ chainId: '0x89', chainRpcUrl: 'https://polygon-rpc.com/' })}
```
Comment (katspaugh) [inline packages/demo/src/App.svelte:272]: I couldn't find any tests in this repo, so I updated the demo. It won't do anything unless the version of the core package is published and updated correspondingly in the demo package.json.

Please let me know if there's any test that I could extend instead.

**Comment by katspaugh** [comment]
@aaronbarnardsound @Adamj1232 @taylorjdawson gm! Any feedback for this?

**Comment by aaronbarnardsound** [comment]
Thanks for this PR @katspaugh. Another approach that I think might be more clear naming wise, is to add an optional `publicRpcUrl` parameter to the `Chain` object and use that in the add chain method. That way the `Chain` info can be set in one place and then there's no need to think about it when setting the chain. I also think it will help distinguish between the two `rpcUrl` parameters. What do you think?

**Comment by aaronbarnardsound** [comment]
Just adding a note @taylorjdawson while we are here we should also add an option `blockExplorerUrl` parameter as requested in #948

**Comment by katspaugh** [comment]
Makes sense!

I'll update the PR to extend the Chain object then 👍

**Comment by taylorjdawson** [comment]
Hey @katspaugh! I see some commits here, is this ready for re-review?

**Comment by katspaugh** [comment]
I haven't fully tested this version yet but a review would be appreciated, yes!

**Comment by taylorjdawson** [comment]
@katspaugh This looks good! Have you had a chance to test?

**Comment by katspaugh** [comment]
Still wrapping my head around testing monorepos, I'll get back to you on the weekend!

**Comment by taylorjdawson** [comment]
@katspaugh I can also test if you want, was there anything else you wanted to add to the PR?

**Comment by katspaugh** [comment]
Oh, that'd be awesome. Let me just fix those git conflicts in the package.json's.

**Review by mahmud-bn — APPROVED** [review]

**Comment by mahmud-bn** [comment]
Hi @katspaugh 
Thanks for creating the PR. This has been merged

**Comment by katspaugh** [comment]
Amazing, thank you guys!

**Pull request merged**